default.conf

```diff
@@ -0,0 +1,26 @@
+server {
+    listen       80;
+    listen  [::]:80;
+    server_name  localhost;
+
+    #access_log  /var/log/nginx/host.access.log  main;
+    # rewrite_log on;
+
+    root /usr/share/nginx/html;
+    index index.html index.htm;
+
+    location / {
+        try_files $uri $uri/ /index.html;
+    }
+
+    # redirects any sub-routes to root for anything containing extension except html
+    rewrite ([^\/]*)\.(?!html)(.*)$ /$1.$2;
```

Comment:
This is really hobbled together; it does the same thing as your app.use in ./app.js where it translates all requests for whatever/whatever.ext to just ./whatever.ext. Except for html, which is handled in the location / thing.